lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2522,7 +2549,11 @@ def bar3d(self, x, y, z, dx, dy, dz, color=None,
                 facecolors *= (6 * len(x))
 
         normals = self._generate_normals(polys)
```

Comment:
This can also go into the if clause, since the normals aren't needed if there aren't any shading